vllm/entrypoints/openai/protocol.py

```diff
@@ -478,13 +478,13 @@ def check_tool_usage(cls, data):
             # it matches a valid tool
             if isinstance(data["tool_choice"], dict):
                 valid_tool = False
-                specified_function = data["tool_choice"]["function"]
+                specified_function = data["tool_choice"].get("function")
                 if not specified_function:
                     raise ValueError(
                         "Incorrectly formatted `tool_choice`. Should be like "
                         "`{\"type\": \"function\","
                         " \"function\": {\"name\": \"my_function\"}}`")
```

Comment:
I think we should have different error message for this case.

Reply:
Thanks for the comment. I fixed this in the new revision.

Reply:
This sounds like the user didn't specify a function, rather than the function not existing server side.

Reply:
Yes, it's more clear now.